### DISCUSSION TITLE
Connect/disconnect OpenVPN client

### DISCUSSION THREAD

**Discussion by PacmanForever** [Vaskivskyi/ha-asusrouter#879] [General]
Hi everyone,

Is it possible to connect and disconnect an OpenVPN client from Home Assistant? I don't see any service that allows me to do that.

Thanks

**Comment by Vaskivskyi** [comment]
Hello,

If we are talking about the in-built OpenVPN Clients - they can be turned on / off already (works with FW 386 and 388 currently and support for 3.0.0.6.102 is in work)

If we are talking about a client connected to the in-built OpenVPN Server - no.

But, in principle, AsusRouter potentially can do almost everything that the WebUI of the device can do. So, different new features can possibly be added. I just need to fully understand what exactly is required and how does it work from the WebUI point of view

**Reply by PacmanForever** [reply]
Yes, I talking about OpenVPN Clients, but I don't see this option in Home Assistant integration. This is my doubt.

My router firmware is 3.0.0.4.386_51679, is the last for the model.

**Comment by maxyge** [comment]
Hello, I have an Asus DSL-AX82U 3.0.0.4.388_24906, and I installed this amazing integration on Home Assistant. However, the switches to activate the OpenVPN clients are all showing as unknown.
Do you have any suggestions?

**Comment by PacmanForever** [comment]
I don't see the option to control VPN clients in Home Assistant...

**Reply by maxyge** [reply]
![Immagine 2024-12-15 122442](https://github.com/user-attachments/assets/b8a83b39-b354-49ff-91d6-49003488167c)
It is a hidden entity, you need to enable it.

**Comment by PacmanForever** [comment]
Does not appear...

![image](https://github.com/user-attachments/assets/583f11df-21a2-44a8-9c53-81b5eb05f871)

**Comment by Vaskivskyi** [comment]
Hello. I am really sorry for such a long delay to get back to you. I did not have time to work on AsusRouter lately due to some other obligations.

@PacmanForever, this is definitely a bug, since your device (as far as I know) should support OpenVPN client switches. Please, open an issue so that I will be able to monitor it properly and we can do some troubleshooting to fix it.

@maxyge, can you please also open a separate issue. We'll try to troubleshoot it. Usually, unknown state for OVPN switches means they are not set up properly the (client does not actually exist - then the library cannot check the state) or some format is different for your device.

Allow me to check the issue till the end of the week (Jan 19, 2025). If there won't be any update from my side till that time, please, tag me again here using `@vaskivskyi` (without code quotes)
